### ISSUE TITLE
[java] Using val causes the error unused private method

### ISSUE THREAD

**Issue by aruncollections** [pmd/pmd#1999]
**Affects PMD Version:** 6.17.0 

**Rule:** UnusedPrivateMethod

**Description:**
Using Lombok's val causes this rule to fail though the private method is used in the expression like below. 

If the following code is used instead of Lombok's val, then it works fine.

String someValue = aPrivateMethod();

**Code Sample demonstrating the issue:**

```
val someValue = aPrivateMethod();

private String aPrivateMethod() {
     return "result";
}
```

**Running PMD through:** *[CLI | Ant | Maven | Gradle | Designer | Other]*
I use Gradle.

**Comment by jsotuyod** [comment]
@aruncollections I can't reproduce the issue using this test-case code.

```xml
    <test-code>
        <description>Lombok.val should not fail</description>
        <expected-problems>0</expected-problems>
        <code><![CDATA[
import lombok.val;

public class ValExample {
  public String example() {
    val example = aPrivateMethod();
    return example;
  }
  
  private String aPrivateMethod() {
   return "result";
  }
}
        ]]></code>
    </test-code>
```
